examples/main.rs

```diff
@@ -16,57 +20,104 @@ fn main() {
     println!("ColorType is {:?}", rgb_img.color());
 
     let mut rgba_img = ImageBuffer::new(width, height);
+    let mut rg_img = ImageBuffer::new(width, height);
+    let mut r_img = ImageBuffer::new(width, height);
 
-    println!("Converting RGB -> RGBA"); // could be optimized
+    println!("Converting RGB -> RGBA/RG/R"); // could be optimized
     for x in 0u32..width {
         for y in 0u32..height {
             let pixel = rgb_img.get_pixel(x, y);
             let pixel_rgba = pixel.to_rgba();
+            let pixel_rg = LumaA::from([pixel_rgba[0], pixel_rgba[1]]);
+            let pixel_r = Luma::from([pixel_rgba[0]]);
             rgba_img.put_pixel(x, y, pixel_rgba);
+            rg_img.put_pixel(x, y, pixel_rg);
+            r_img.put_pixel(x, y, pixel_r);
         }
     }
 
     let block_count = intel_tex_2::divide_up_by_multiple(width * height, 16);
     println!("Block count: {}", block_count);
-
-    let mip_count = 1;
-    let array_layers = 1;
-    let caps2 = Caps2::empty();
-    let is_cubemap = false;
-    let resource_dimension = D3D10ResourceDimension::Texture2D;
-    let alpha_mode = AlphaMode::Opaque;
-    let depth = 1;
-
-    let mut dds = Dds::new_dxgi(NewDxgiParams {
+    let dds_defaults = NewDxgiParams {
         height,
         width,
-        depth: Some(depth),
+        depth: Some(1),
         format: DxgiFormat::BC7_UNorm,
-        mipmap_levels: Some(mip_count),
-        array_layers: Some(array_layers),
-        caps2: Some(caps2),
-        is_cubemap,
-        resource_dimension,
-        alpha_mode,
-    })
-    .unwrap();
-
-    let surface = intel_tex_2::RgbaSurface {
-        width,
-        height,
-        stride: width * 4,
-        data: &rgba_img,
+        mipmap_levels: Some(1),
+        array_layers: Some(1),
+        caps2: Some(Caps2::empty()),
+        is_cubemap: false,
+        resource_dimension: D3D10ResourceDimension::Texture2D,
+        alpha_mode: AlphaMode::Opaque,
     };
+    // BC4
+    {
+        let surface = intel_tex_2::RSurface {
+            width,
+            height,
+            stride: width,
```

Comment:
Shouldn't we check `pitch` here in case every scan-line has padding?

https://docs.rs/ddsfile/latest/ddsfile/struct.Dds.html?search=#method.get_pitch

Reply:
Hmm that would be the pitch of the DDS file that we're writing. Should the input data already be corrected for that? If so we'll need to create an `ImageBuffer` that accounts for that and make sure our compressor (via the field that this review comment is on) accounts for that?

Reply:
After discussing it in person we came up with the solution in [bbbc26e](https://github.com/Traverse-Research/intel-tex-rs-2/pull/25/commits/bbbc26e89cf1edb3a8419879cfaa67306113277f)